config/config_mgmt.py

```diff
@@ -530,12 +530,15 @@ def deleteHandler(diff, inp, outp, config):
             # if output is dict, delete keys from config
             if isinstance(inp, dict):
                 for key in diff:
-                    #print(key)
                     # make sure keys from diff are present in inp but not in outp
                     # then delete it.
                     if key in inp and key not in outp:
-                        # assign key to None(null), redis will delete entire key
-                        config[key] = None
+                        if type(inp[key]) == list:
```

Comment:
What are the valid types today?

Reply:
This is done for both fields and KEYs which can be deleted while DPB, today we hit here only for KEY[Dict] and for fields which are list. so 2 types dict and list.

3rd type can be String,  we do not have any such case for DPB now. But in future, if needed, Such field can be set to "" or "None". As of now, it is None. This will again force us to add default values in YANG models.

Reply:
Since the backend is not doing delete anyways,  I was wondering why we separate list vs other types here?

Reply:
This code handled keys as well, so we need to write say "VLAN100|Ethernet4": None. Which is handled by backhand and it removes the key.
Same code handles fields as well, because from Python POV everything on left side is Key. For fields, None will be mapped to "None"--> Like "FEC": "None". We have no case which hits this as of now while DPB.
To differentiate b/w ConfigDB Key and field, I need to pass the level to this function. It can be done if needed, so we assign KEY as None. And Field as either [] or "".